Feat: Add Iota Demo

bhelx opened the pull request:
Closes #65 


Adds the demo and auto-deploys to northflank on each merge. After this is merged i'll point northflank to main. Right now it's pointed to this branch.

bhelx commented on demo-iota/rust/Cargo.toml:
```diff
@@ -0,0 +1,20 @@
+wasi-common = "8.0.0"
+wasmtime = "8.0.0"
+wasmtime-wasi = "8.0.0"
+dylibso-observe-sdk = { git = "https://github.com/dylibso/observe-sdk", branch = "main" }
```
I need to change this to use path. but it's a little tricky

bhelx commented on demo-iota/go/go.mod:
```diff
@@ -0,0 +1,13 @@
+go 1.19
+
+require (
+	github.com/dylibso/observe-sdk/go v0.0.0-20230627221207-8daea0862c34
```
Same here. this should link to the local codebase if possible

nilslice commented:
Should we include the language suffix on the service names? only thinking it would make determining which of the adapters we're testing a bit easier. Unless there's another way to know which service/adapter the request came in through from the APM UIs?

bhelx commented:
> Should we include the language suffix on the service names? only thinking it would make determining which of the adapters we're testing a bit easier. Unless there's another way to know which service/adapter the request came in through from the APM UIs?

I think it makes sense to keep it in the same service. we can aggregate across them, etc. I added a [default tag](https://github.com/dylibso/observe-sdk/pull/69/commits/521b5e9f6b07e9d1c8dea90c3cc8234b614524ce) called `host_language` to each adapter. So you can use `@host_language:rust` in your queries for example.  I also added it as a column in the trace list, and a filterable facet so it's easy to identify them and filter them out, or do computation or graphs on them

<img width="1343" alt="Screenshot 2023-08-03 at 8 25 15 AM" src="https://github.com/dylibso/observe-sdk/assets/185919/ce4b1e12-8976-44e1-9451-1a50dfdfb84d">

bhelx commented:
I need to follow up and add the metadata to both of these. Will have a ticket.
